azure-client-runtime/src/test/java/com/microsoft/azure/v2/AzureProxyTests.java

```diff
@@ -515,6 +526,76 @@ public void accept(OperationStatus<MockResource> operationStatus) {
         assertEquals(3, httpClient.pollRequests());
     }
 
+    @Test
+    public void beginAndResumeCreateAsyncWithLocationAndPolls() {
+        final MockAzureHttpClient httpClient = new MockAzureHttpClient();
+
+        final AtomicInteger inProgressCount = new AtomicInteger();
+        final Value<MockResource> resource = new Value<>();
+        final StringBuffer data = new StringBuffer();
+        final ObjectMapper mapper = new ObjectMapper();
+        mapper.setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE);
+        mapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+        mapper.enableDefaultTyping();
+
+        createMockService(MockResourceService.class, httpClient)
+                .beginCreateAsyncWithLocationAndPolls("1", "mine", "c", 10)
+                .take(2)
+                .subscribe(new Consumer<OperationStatus<MockResource>>() {
+                    @Override
+                    public void accept(OperationStatus<MockResource> operationStatus) {
+                        if (!operationStatus.isDone()) {
+                            OperationDescription operationDescription = operationStatus.buildDescription();
+                            try {
+                                data.append(mapper.writeValueAsString(operationDescription));
+                            } catch (JsonProcessingException e) {
+                                fail("Error serializing OperationDescription object");
+                                e.printStackTrace();
+                            }
+                            inProgressCount.incrementAndGet();
+                        }
+                        else {
+                            resource.set(operationStatus.result());
+                        }
+                    }
+                });
+
+        OperationDescription operationDescription = null;
+        PollStrategy.PollStrategyData pollData = null;
+        try {
+            operationDescription = mapper.readValue(data.toString(), OperationDescription.class);
```

Comment:
Are you able to support a use case/add test coverage where an OperationDescription is assembled from user-provided values? One of the main users requesting this feature has asked that they be able to store the information received from a begin* method in whatever format they wish and convert it themselves to the format we need to resume polling.

Reply:
A user would need to start a begin operation and then serialize the data to get the values for resuming the observing of the op.  The serialization to a JSON string would cover the case of manually supplying values as the JSON the user built up would need to be deserialized back to the OperationDescription object.